### ISSUE TITLE
Add orchestrator default location not under `src`

### ISSUE THREAD

**Issue by weikanglim** [dotnet/aspire#1422]
VS version:
Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.9.0 Preview 2.0

Repro:

1. Right click on a project located at `src/MyProject/Project.csproj`. Select "Add .NET Aspire Orchestrator Support" in Visual Studio.
2. Accept defaults. This creates the projects not under `src`, but directly under the solution directory, `MySolution.AppHost/MySolution.AppHost.csproj`
3. Run `azd init` in the solution folder. The project fails to initialize correctly, since a `src` directory exists and `azd` assumes all projects are under `src`.

**Comment by weikanglim** [comment]
`azd` could be more flexible in its heuristics as well. Currently, if `src` directory exists, we only scan `src` directory because we assume all source code would most likely be under `src`.

**Comment by DamianEdwards** [comment]
In the upcoming release, the directory the projects are created in can be changed in the dialog:

<img width="316" alt="image" src="https://github.com/dotnet/aspire/assets/249088/b122dd51-599b-4183-90f5-f8388c852385">

Are you asking for the default directory to special-case `src` if the project that was clicked on is in a directory named `src` to start with?

@phenning

**Comment by weikanglim** [comment]
> Are you asking for the default directory to special-case src if the project that was clicked on is in a directory named src to start with?

Indeed, if the project being added is `<solution_dir>/src`, default to `<solution_dir/src` instead of `<solution_dir>`.

The main motivation that increases the priority of this is that `azd` currently special-cases the `src` directory to avoid false positives in the non-aspire, general app detection case. This means if a user blindly accept defaults in this particular case, they may get unexpected behavior described above.

We can also make a tradeoff to make the general detection looser to make this off-path case better handled, so that's always an option on the table. Just wanted to start the discussion to see what we prefer for the end-to-end experience.

**Comment by davidfowl** [comment]
@phenning close if it is no longer relevant

**Comment by davidfowl** [comment]
cc @vijayrkn

**Comment by phenning** [comment]
We had the customizable path for some time without the defaulting to 'src'. Hoping to have that in for 17.10

**Comment by phenning** [comment]
This should be in for Preview 3
